lib/session.js

```diff
@@ -20,6 +20,49 @@ var hash = function(str) {
 };
 
 /**
+ * Signs in a user using user obejct
+ *
+ * @param {Object} user - user object
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @param {function()} onSuccess callback, is passed the User instance
+ */
+
+exports.signinWithUser = function signinWithUser(user, req, res, onSuccess) {
+	if (arguments.length < 4) {
+		console.error('\nkeystone.sesson.signinWithUser requires user, req and res objects, and an onSuccess callback.\n');
+		process.exit(1);
```

Comment:
process.exit is really unsafe, and only something I'm comfortable using in keystone config validation - not code that would be run in the normal context of application execution.

There's no reason to kill the process here, it can just throw an error.